commander/rollup.go

```diff
@@ -61,6 +65,13 @@ func (c *Commander) rollupLoop(ctx context.Context) (err error) {
 			if err != nil {
 				return err
 			}
+		case <-updateMempoolTicker.C:
+			// TODO: a long rollupLoopIteration can starve this metric,
+			//       create a separate worker
```

Comment:
Even worse would be a long `updateMempoolMetric` starving the `rollupLoopItteratoin`.

Reply:
The update is O(n) in mempool size, so in theory this is possible. In such a case the mempool would keep growing without rollupLoopItteration getting called enough to shrink it.

But the update looks like it should be really fast, so extremely unlikely to happen in practice. And if it does, we will have nice up-to-date metrics to see it happening :)

Reply:
Ah, I'm embarrassed to say I didn't consider the other direction.

I agree it seems unlikely for now, but that kind of phase transition is scary enough that it bumps up the priority of moving this into a new worker: https://linear.app/worldcoin/issue/PRO-310/move-mempool-metric-into-separate-worker